src/gtkspellcheck/spellcheck.py

```diff
@@ -149,32 +151,110 @@ def exists(self, language):
             return language in self.mapping
 
     class _Mark:
-        def __init__(self, buffer, name, start):
+        def __init__(self, buffer, name, start, iter_worker):
             self._buffer = buffer
             self._name = name
             self._mark = self._buffer.create_mark(self._name, start, True)
+            self._iter_worker = iter_worker
 
         @property
         def iter(self):
             return self._buffer.get_iter_at_mark(self._mark)
 
         @property
         def inside_word(self):
-            return self.iter.inside_word()
+            return self._iter_worker.inside_word(self.iter)
 
         @property
         def word(self):
             start = self.iter
-            if not start.starts_word():
-                start.backward_word_start()
+            if not self._iter_worker.starts_word(start):
+                self._iter_worker.backward_word_start(start)
             end = self.iter
-            if end.inside_word():
-                end.forward_word_end()
+            if self._iter_worker.inside_word(end):
+                self._iter_worker.forward_word_start(end)
             return start, end
 
         def move(self, location):
             self._buffer.move_mark(self._mark, location)
 
+    class _IterWorker:
+        def __init__(self, extra_word_chars):
+            self._extra_word_chars = extra_word_chars
+
+        def is_extra_word_char(self, loc):
+            # Language extra chararacters should also be processed once Enchant's
+            # enchant_dict_get_extra_word_characters is exposed in PyEnchant
+
+            return loc.get_char() in list(self._extra_word_chars)
```

Comment:
The `list` is unnecessary here. You can just do `"x" in "xyz"`.

Not sure whether it makes sense to transform `_extra_word_chars` into a `set`.

Reply:
`list` is used to ensure `get_char` returning an empty string won't return positive

Having `_extra_word_chars` as a list not a string feels cleaner to me but with my low budget of time at the moment didn't come across an easy way to expose that in the GObject property interface.

Reply:
I see. That makes sense indeed. Sorry for dismissing it as unnecessary too quickly.

Still, creating a list from scratch all the time is very inefficient. Can we just add this case to the `if`:
```python
char = loc.get_char()
return char and char in self._extrac_word_chars
```
This should avoid additional allocations for the list and all that.

Reply:
No worries, and yes completely agree with avoiding creating the list every time.

`Gtk.TextIter.get_char` returns an empty string at the end of the buffer, which with your suggestion would result in an empty string being returned from the method instead of a bool, so I've put this in instead

```python
char = loc.get_char()
return char != '' and char in self._extra_word_chars
``` 

Naturally there are plenty of ways to do this, if you want a separate end iterator check instead let me know.